help_to_heat/frontdoor/schemas.py

```diff
@@ -853,6 +857,20 @@ class SessionSchema(Schema):
     epc_found = fields.String()
     epc_rating_is_eligible = fields.String()
 
+    @validates("postcode")
+    def validate_postcode(self, value):
+        if value:
+            value = value.upper().replace(" ", "")
+            valid_postcode = False
+
+            for valid_pattern in postcode_regex_collection:
+                if re.match(valid_pattern, value):
+                    valid_postcode = True
+                    break
+
+            if not valid_postcode:
+                raise ValidationError([_("Enter a valid UK postcode")])
```

Comment:
Could condense this and make it a little clearer by inverting the check:
```suggestion
            if not any(map(lambda pattern: re.match(pattern, value), postcode_regex_collection)):
                raise ValidationError([_("Enter a valid UK postcode")])
```

Reply:
Yes was going to say there was probably a more Python way to write the same thing